javatests/arcs/android/systemhealth/testapp/StorageCore.kt

```diff
@@ -576,6 +576,14 @@ class StorageCore(val context: Context, val lifecycle: Lifecycle) {
             }
 
             notify { "Progress 100%: populating stats and report" }
+            // Close all Handles and EntityHandleManagers
+            handles.forEach {
+                runBlocking {
+                    (it.handle as Handle).close()
+                    it.handleManager.close()
```

Comment:
The handleManager.close() should be moved to the function:
`private suspend inline fun <T> closeHandleSuspend(handle: T?)`

Reply:
I tried this and it doesn't work. I'm guessing either terminated() isn't always invoked, it is invoked after the stats are populated?

Reply:
it's in the block which is run after generating hprof file.
```
finally {
     // Gracefully shut all tasks down which in turns calls all tasks' terminated().
     tasks.forEach { it.shutdown() }
}
```
The shutdown will cause the terminated() getting called.

Reply:
Let's get your version in first :-)
I'd like to refactor the tear-down flow a little bit when I get some spare cycles then move the handleManager.close() to where handle.close() is.